spec/src/main/doc/core/injectionandresolution.asciidoc

```diff
@@ -346,23 +344,17 @@ The behavior of all methods declared by +java.lang.Object+, except for +toString
 
 === Dependency injection
 
-From time to time the container instantiates beans and other Java EE component classes supporting injection. The resulting instance may or may not be a _contextual instance_ as defined by <<contextual_instance>>.
+From time to time the container instantiates beans and other class supporting injection. The resulting instance may or may not be a _contextual instance_ as defined by <<contextual_instance>>.
 
-The container is required to perform dependency injection whenever it creates one of the following contextual objects:
+The container is required to perform dependency injection whenever it creates the following contextual objects:
 
-* contextual instances of session beans, and
 * contextual instances of managed beans.
 
+The container is also required to perform dependency injection whenever it instantiates the following non-contextual objects:
 
-The container is also required to perform dependency injection whenever it instantiates any of the following non-contextual objects:
-
-* non-contextual instances of session beans (for example, session beans obtained by the application from JNDI or injected using +@EJB+),
-* non-contextual instances of managed beans, and
-* instances of any other Java EE component class supporting injection.
-
-A Java EE 5 container is not required to support injection for non-contextual objects.
+* non-contextual instances of managed beans.
 
-The container interacts with instances of beans and other Java EE component classes supporting injection by calling methods and getting and setting field values.
+The container interacts with instances of beans or objects supporting injection by calling methods and getting and setting field values.
```

Comment:
In CDI core these objects were instantiated via `UnManaged` or an `InjectionTarget`, so I think it's ore object than classes.